azurelinuxagent/distro/default/rdma.py

```diff
@@ -0,0 +1,74 @@
+# Windows Azure Linux Agent
+#
+# Copyright 2016 Microsoft Corporation, SUSE LLC, Robert Schweikert
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+Handle packages and modules to enable RDMA for IB networking
+"""
+
+import os
+import azurelinuxagent.logger as logger
+
+
+class RDMAHandler(object):
+    def __get_rdma_version(self):
+        """Retrieve the firmware version information from the system.
+           This depends on information provided by the Linux kernel."""
+
+        driver_info_source = '/var/lib/hyperv/.kvp_pool_0'
+        base_kernel_err_msg = 'Kernel does not provide the necessary '
+        base_kernel_err_msg += 'information or the hv_kvp_daemon is not '
+        base_kernel_err_msg += 'running.'
+        if not os.path.exists(driver_info_source):
```

Comment:
a `os.path.isfile` check would be better. it contains the `exists` check as well.